Tests/MapboxMapsTests/Ornaments/Compass/CompassMapViewIntegrationTests.swift

```diff
@@ -0,0 +1,57 @@
+import XCTest
+#if canImport(MapboxMaps)
+@testable import MapboxMaps
+#else
+@testable import MapboxMapsOrnaments
+#endif
+
+class CompassMapViewIntegrationTests: MapViewIntegrationTestCase {
+
+    func testUpdateMapBearing() throws {
+        let mapView = try XCTUnwrap(self.mapView, "Map view could not be found")
+
+        let initialSubviews = mapView.subviews.filter { $0.isKind(of: MapboxCompassOrnamentView.self) }
+
+        let compass = try XCTUnwrap(initialSubviews.first as? MapboxCompassOrnamentView, "The MapView should include a compass view as a subview")
+
+        XCTAssertEqual(mapView.mapboxMap.cameraState.bearing, 0, "The map's initial bearing should be equal to 0")
+        XCTAssertTrue(compass.containerView.isHidden, "The compass should be hidden initially")
+        XCTAssertEqual(mapView.mapboxMap.cameraState.bearing, compass.currentBearing, "The map's initial bearing should be equal to the compass' bearing")
+
+        mapView.camera.setCamera(to: CameraOptions(bearing: 30))
+        XCTAssertFalse(compass.containerView.isHidden, "The compass should hidden when the bearing is 30.")
+        XCTAssertEqual(mapView.mapboxMap.cameraState.bearing, compass.currentBearing, "The map's bearing should be equal to the compass' current bearing")
+        XCTAssertEqual(mapView.mapboxMap.cameraState.bearing, 30, accuracy: 0.2, "The map's bearing should be equal to 30 with an accuracy of 0.2.")
+
+        mapView.camera.setCamera(to: CameraOptions(bearing: 0))
+        XCTAssertTrue(compass.containerView.isHidden)
+        XCTAssertEqual(mapView.mapboxMap.cameraState.bearing, 0)
+    }
+
+    func testCompassTappedResetsToNorth() throws {
+        let mapView = try XCTUnwrap(self.mapView, "Map view could not be found")
+
+        let initialSubviews = mapView.subviews.filter { $0.isKind(of: MapboxCompassOrnamentView.self) }
+
+        let compass = try XCTUnwrap(initialSubviews.first as? MapboxCompassOrnamentView, "The MapView should include a compass view as a subview")
+
+        mapView.camera.setCamera(to: CameraOptions(bearing: 30))
```

Comment:
Would an `ease` with a completion block work here? Worried with the `onEvery` that it could capture going *through* 0.

Reply:
That's fair, the only issue would be that we are trying to capture the result after `compassTapped()`, not after `setCamera`. The camera animations after `compassTapped` take around 0.3 seconds. We should improve this test as we add more ornaments tests, though.

Reply:
Yeah, sorry, I'm being a dunce.